### ISSUE TITLE
Sync XHR deprecated on the main thread

### ISSUE THREAD

**Issue by ridomin** [ripple-emulator/ripple#19]
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org

**Comment by axemclion** [comment]
@ridomin I agree - would be great if you can send in a pull request that fixes this issue.
